artifacts.go

```diff
@@ -71,39 +70,51 @@ func read(aPath string) (parser.Workers, error) {
 	if ar == nil {
 		return nil, errors.New("Can not read artifact file.")
 	}
-	defer ar.Close()
 
 	p := parser.RootfsParser{
 		W: ioutil.Discard, // don't store update anywhere
 	}
 	ar.Register(&p)
 
-	return ar.Read()
+	_, err = ar.Read()
+	if err != nil {
+		return nil, err
+	}
+	defer ar.Close()
+
+	return ar, nil
```

Comment:
This a bit confusing: Are you throwing away the result of the first `Read()`, and then returning the object? I'm unsure if you're doing the right thing here. At least a comment to explain would be nice.